python/thunder/rdds/series.py

```diff
@@ -636,6 +636,111 @@ def query(self, inds, var='inds', order='F', onebased=True):
 
         return keys, values
 
+    def toBlocks(self, blockSizeSpec="150M"):
+        """
+        Parameters
+        ----------
+        blockSizeSpec: string memory size, tuple of integer splits per dimension, or instance of BlockingStrategy
+            A string spec will be interpreted as a memory size string (e.g. "64M"). The resulting blocks will be
+            generated by a SeriesBlockingStrategy to be close to the requested size.
+            A tuple of positive ints will be interpreted as "splits per dimension". Only certain patterns of splits
+            are valid to convert Series back to Blocks; see docstring above. These splits will be passed into a
+            SeriesBlockingStrategy that will be used to generate the returned blocks.
+            If an instance of SeriesBlockingStrategy is passed, it will be used to generate the returned Blocks.
+
+        Returns
+        -------
+        Blocks instance
+        """
+        from thunder.rdds.imgblocks.strategy import BlockingStrategy, SeriesBlockingStrategy
+        if isinstance(blockSizeSpec, SeriesBlockingStrategy):
+            blockingStrategy = blockSizeSpec
+        elif isinstance(blockSizeSpec, basestring) or isinstance(blockSizeSpec, int):
+            blockingStrategy = SeriesBlockingStrategy.generateFromBlockSize(self, blockSizeSpec)
+        else:
+            # assume it is a tuple of positive int specifying splits
+            blockingStrategy = SeriesBlockingStrategy(blockSizeSpec)
+
+        blockingStrategy.setSource(self)
+        avgSize = blockingStrategy.calcAverageBlockSize()
+        if avgSize >= BlockingStrategy.DEFAULT_MAX_BLOCK_SIZE:
+            # TODO: use logging module here rather than print
+            print "Thunder WARNING: average block size of %g bytes exceeds suggested max size of %g bytes" % \
+                  (avgSize, BlockingStrategy.DEFAULT_MAX_BLOCK_SIZE)
+
+        returnType = blockingStrategy.getBlocksClass()
+        blockedRdd = self.rdd.map(blockingStrategy.blockingFunction)
+        # since our blocks are likely pretty big, try setting 1 partition per block
+        groupedRdd = blockedRdd.groupByKey(numPartitions=blockingStrategy.nblocks)
+        # <key>, <val> at this point is:
+        # <block number>, <[(series key, series val), (series key, series val), ...]>
+        simpleBlocksRdd = groupedRdd.map(blockingStrategy.combiningFunction)
+        return returnType(simpleBlocksRdd, dims=self.dims, nimages=len(self.index), dtype=self.dtype)
+
+    def saveAsBinarySeries(self, outputdirname, overwrite=False):
```

Comment:
I've been thinking of `Binary Series` as *almost* a separate data type from a regular `Series`. In the file output code, series partitions first get mapped from a format where a single record is a 1d array to one where records are these big concatenated binary blobs of keys and values for the whole partition. This second blob representation (plus metadata) I've been thinking of as a "binary series". This is as opposed to, say, writing out a series as a sequence file, where the sequence file abstraction lets you think of the data as being serialized with their format relatively unchanged.

So from that point of view where "binary series" is a thing, it's not redundant. :)